frontend/src/Buildings/buildingReducer.js

```diff
@@ -0,0 +1,20 @@
+import { FETCH_BUILDINGS_START, FETCH_BUILDINGS_SUCCESS, FETCH_BUILDINGS_ERROR } from './actionCreator';
+
+const initialState = {
+  buildingsInfo: [],
+  isLoading: false,
+  error: '',
+};
+
+export default function buildings(state = initialState, action) {
+  switch (action.type) {
+    case FETCH_BUILDINGS_START:
+      return { ...state, isLoading: true };
+    case FETCH_BUILDINGS_SUCCESS:
+      return { ...state, buildingsInfo: action.payload, isLoading: false };
+    case FETCH_BUILDINGS_ERROR:
+      return { ...state, error: action.payload, isLoading: false };
```

Comment:
please keep the order of the initialState for keeping it easy to read, so:
```suggestion
      return { ...state, isLoading: false, error: action.payload };
```
buildingsInfo, isLoading, error